projects/rocketpool/index.js

```diff
@@ -1,156 +1,66 @@
-  const unmatched_minipools = minipool_count_per_status[0] * 16 // Unmatched minipools
-  const pending_minipools = minipool_count_per_status[1] * 32 // Pending minipools (matched but not staking yet)
-  const staking_minipools = minipool_count_per_status[2] * 32 // Staking minipools
-  const withdrawable_minipools = minipool_count_per_status[3] * 32 // Withdrawable minipools
-  // Deposit pool balance
-  // rocketDepositPool_balance = solidity.to_float(rp.call("rocketDepositPool.getBalance"))
-  // rETH collateral from withdrawn minipools
-  // rETH_collateral_from_withdrawn_minipools = solidity.to_float(w3.eth.getBalance(rp.get_address_by_name("rocketTokenRETH")))
+  const addresses = await api.call({ target: rocketNodeManager, abi: abi.getNodeAddresses, params: [0, nodeLength] });
 
-  const ETH_TVL = staking_minipools
-          + pending_minipools
-          + unmatched_minipools
-          + withdrawable_minipools
-          + parseFloat(rocketDepositPoolBalance) / 1e18
-          + parseFloat(rocketTokenRETHBalance) / 1e18
-
-  // RPL staked
-  // rpl_tvl += "rocketNodeStaking.getTotalRPLStake")) // RPL staked by Node Operators
-  // rpl_tvl += solidity.to_float(rp.call("rocketVault.balanceOfToken", "rocketDAONodeTrustedActions", rpl)) // RPL bonded by the oDAO
-  // rpl_tvl += solidity.to_float(rp.call("rocketVault.balanceOfToken", "rocketAuctionManager", rpl)) // slashed RPL that hasn't been auctioned off yet
-  const RPL_tvl = parseFloat(totalRPLStake) + parseFloat(rocketDAONodeTrustedActions_rplBalance) + parseFloat(rocketAuctionManager_rplBalance)
-
-// pending_minipools: ${pending_minipools}
-// unmatched_minipools: ${unmatched_minipools}
-// withdrawable_minipools: ${withdrawable_minipools}
-// rocketDepositPoolBalance: ${rocketDepositPoolBalance / 1e18}
-// rocketTokenRETHBalance: ${rocketTokenRETHBalance / 1e18}
-// = ETH_TVL: ${ETH_TVL}\n
-// rocketNodeStaking.getTotalRPLStake: ${totalRPLStake/1e18}
-// rocketDAONodeTrustedActions_rplBalance: ${rocketDAONodeTrustedActions_rplBalance/1e18}
-// rocketAuctionManager_rplBalance: ${rocketAuctionManager_rplBalance/1e18}
-// = RPL_tvl: ${RPL_tvl/1e18}\n`) 
-
-  const balances = {
-    [weth]: ETH_TVL * 1e18, 
-    [rpl]: RPL_tvl
+  const batchSize = 100;
+  const batchedAddresses = [];
+  for (let i = 0; i < addresses.length; i += batchSize) {
+    batchedAddresses.push(addresses.slice(i, i + batchSize));
   }
-  return balances
-}
 
-module.exports = {
-  methodology: "Rocketpool TVL is ethereum staked by the users and node operators - collateral provided against rETH - staked on beacon chain 32 * activeMinipoolCount + RPL staked by Node Operators to operate a node.",
-  ethereum: {
-    tvl,
-  },
+  const nodeDetails = await Promise.all(
+    batchedAddresses.map(async (batch) => {
+      const results = await api.multiCall({ calls: batch.map((address) => ({ target: rocketNodeManager, params: [address] })), abi: abi.getNodeDetails, permitFailure: true })
```

Comment:
I have added [`excludeFailed`](https://github.com/DefiLlama/defillama-sdk/blob/master/src/types.ts#L41) flag for this